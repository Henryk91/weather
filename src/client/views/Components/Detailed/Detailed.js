import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import ReactAnimatedWeather from 'react-animated-weather';
import { Hourly } from '../index';
// import './style.css';
import { getTime, convertIconName } from '../../Helpers/helper-functions';


export default class Detailed extends Component {
    constructor(props) {
        super(props);
        this.state = {
            dayData: null,
        };
    }

    render() {
        const weatherData = this.props.weatherData
        // console.log(weatherData)
        let dayInfo = this.state.dayData;
        if (this.props.match) {
            dayInfo = weatherData.daily.data.filter((val) => val.time == this.props.match.params.id)[0];
        } else {
            dayInfo = weatherData.daily.data[0]
        }
        let date = new Date(dayInfo.time * 1000)

        return (
            <Link style={{ textDecoration: 'none' }} to="/">
                <h1>{date.toDateString()}</h1>
                <ReactAnimatedWeather 
                    className="detailsIcons"
                    icon={convertIconName(weatherData.currently.icon)}
                    color={'#FEBF34'}
                    size={60}
                    animate={true}
                />

                {this.props.match ? <p className="yellowP">Current Temp: {Math.round(weatherData.currently.temperature)}&#176;</p> : null}
                <p className="yellowP">{dayInfo.summary}</p>
                {this.props.match ? <iframe src={`https://maps.darksky.net/@temperature,${date.toISOString().slice(0,10)},22,${weatherData.latitude},${weatherData.longitude},7?defaultUnits=_c`}></iframe> : null }
                
                <p>(click anywhere to go back)</p>
                <div className="bigScreen detailed">
                    <div>
                        <p className="leadP">Wind</p>
                        <p>Bearing: {dayInfo.windBearing}</p>
                        <p>Gust: {dayInfo.windGust}</p>
                        <p>Gust Time: {getTime(dayInfo.windGustTime)}</p>
                        <p>Speed: {dayInfo.windSpeed}</p>
                    </div>
                    <div>
                        <p className="leadP">Sky</p>
                        <p>Sunrise: {getTime(dayInfo.sunriseTime)} </p>
                        <p>Sunset: {getTime(dayInfo.sunsetTime)}</p>
                        <p>UV Index: {dayInfo.uvIndex}</p>
                        <p>UV Index Time: {getTime(dayInfo.uvIndexTime)}</p>
                        <p>Visibility: {dayInfo.visibility}</p>
                        <p>Ozone: {dayInfo.ozone}</p>
                    </div>
                    <div>
                        <hr />
                        <p className="leadP">Temp</p>
                        <p>High: {dayInfo.temperatureHigh}</p>
                        <p>High Time: {getTime(dayInfo.temperatureHighTime)}</p>
                        <p>Low: {dayInfo.temperatureLow}</p>
                        <p>Low Time: {getTime(dayInfo.temperatureLowTime)}</p>
                        <p>Max: {dayInfo.temperatureMax}</p>
                        <p>Max Time: {getTime(dayInfo.temperatureMaxTime)}</p>
                        <p>Min: {dayInfo.temperatureMin}</p>
                        <p>Min Time: {getTime(dayInfo.temperatureMinTime)}</p>
                    </div>
                    <div>
                        <hr />
                        <p className="leadP">Precipitation</p>
                        <p>Cloud Cover: {dayInfo.cloudCover * 100}</p>
                        <p>Humidity: {dayInfo.humidity}</p>
                        <p>DewPoint: {dayInfo.dewPoint}</p>
                        <p>Intensity: {dayInfo.precipIntensity}</p>
                        <p>Probability: {Math.round(dayInfo.precipProbability * 100)}%</p>
                        <p>Intensity Max: {dayInfo.precipIntensityMax}</p>
                        <p>Intensity Max time: {getTime(dayInfo.precipIntensityMaxtime)}</p>
                        <p>Pressure: {dayInfo.pressure}</p>
                    </div>
                </div>
                {weatherData ? <div><Hourly weatherData={weatherData}/></div> : null}
            </Link>
        );
    }
}
