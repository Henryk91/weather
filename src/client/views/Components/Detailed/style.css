.detailed{
    display:grid;
      grid-template-columns:repeat(2, 1fr);
      grid-auto-rows: auto;
    background: #007998;
    text-align: left;
  }
  .detailed > div{
    margin: 15px;
  }