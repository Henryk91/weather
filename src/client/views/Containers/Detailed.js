import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import ReactAnimatedWeather from 'react-animated-weather';
import { Hourly } from './index';

export default class Detailed extends Component {
    constructor(props) {
        super(props);
        this.state = {
            weatherData: this.props.weatherData,
            // id: this.props.match.params.id,
            dayData: null,
        };
        this.getTime = this.getTime.bind(this)
        this.convertIcon = this.convertIcon.bind(this)
    }
    componentDidMount() {

        let dayData;
        if (this.props.match !== undefined) {
            dayData = this.state.weatherData.daily.data.filter((val) => val.time == this.props.match.params.id)[0];
        } else {
            dayData = this.state.weatherData.daily.data[0]
        }
        this.setState({ dayData })
        console.log(this.state.weatherData)
    }
    convertIcon(icon) {
        return icon.toUpperCase().replace(/-/g, "_")
    }

    getTime(time) {
        var date = new Date(time * 1000)
        var dateTime = date.toTimeString()
        return dateTime.substring(0, dateTime.lastIndexOf(":"))
    }

    render() {


        const data = this.state.weatherData
        
        
        let dayInfo;
        if (this.props.match) {
            dayInfo = this.state.weatherData.daily.data.filter((val) => val.time == this.props.match.params.id)[0];
        } else {
            dayInfo = this.state.weatherData.daily.data[0]
        }
        let date = new Date(dayInfo.time * 1000)
        console.log(date)
        return (
            <Link style={{ textDecoration: 'none' }} to="/">
                <h1>{date.toDateString()}</h1>
                <ReactAnimatedWeather 
                    className="detailsIcons"
                    icon={this.convertIcon(data.currently.icon)}
                    color={'#FEBF34'}
                    size={60}
                    animate={true}
                />
                {/* https://maps.darksky.net/@temperature,2019-3-15,22,-25.428,30.100,8 */}
                {this.props.match ? <p className="yellowP">Current Temp: {Math.round(data.currently.temperature)}&#176;</p> : null}
                <p className="yellowP">{dayInfo.summary}</p>
                {this.props.match ? <iframe src={`https://maps.darksky.net/@temperature,${date.toISOString().slice(0,10)},22,${data.latitude},${data.longitude},7?defaultUnits=auto`}></iframe> : null }
                <p>(click anywhere to go back)</p>
                {/* <div id="quickDay">
                    <div>
                        <p>Morning</p>
                    </div>
                    <div>
                        <p>Noon</p>
                    </div>
                    <div>
                        <p>Night</p>
                    </div>
                </div> */}
                <div className="bigScreen detailed">
                    <div>
                        <p className="leadP">Wind</p>
                        <p>Bearing: {dayInfo.windBearing}</p>
                        <p>Gust: {dayInfo.windGust}</p>
                        <p>Gust Time: {this.getTime(dayInfo.windGustTime)}</p>
                        <p>Speed: {dayInfo.windSpeed}</p>
                    </div>
                    <div>
                        <p className="leadP">Sky</p>
                        <p>Sunrise: {this.getTime(dayInfo.sunriseTime)} </p>
                        <p>Sunset: {this.getTime(dayInfo.sunsetTime)}</p>
                        <p>UV Index: {dayInfo.uvIndex}</p>
                        <p>UV Index Time: {this.getTime(dayInfo.uvIndexTime)}</p>
                        <p>Visibility: {dayInfo.visibility}</p>
                        <p>Ozone: {dayInfo.ozone}</p>
                    </div>

                    <div>
                        <hr />
                        <p className="leadP">Temp</p>
                        <p>High: {dayInfo.temperatureHigh}</p>
                        <p>High Time: {this.getTime(dayInfo.temperatureHighTime)}</p>
                        <p>Low: {dayInfo.temperatureLow}</p>
                        <p>Low Time: {this.getTime(dayInfo.temperatureLowTime)}</p>
                        <p>Max: {dayInfo.temperatureMax}</p>
                        <p>Max Time: {this.getTime(dayInfo.temperatureMaxTime)}</p>
                        <p>Min: {dayInfo.temperatureMin}</p>
                        <p>Min Time: {this.getTime(dayInfo.temperatureMinTime)}</p>
                    </div>

                    <div>
                        <hr />
                        <p className="leadP">Precipitation</p>
                        <p>Cloud Cover: {dayInfo.cloudCover * 100}</p>
                        <p>Humidity: {dayInfo.humidity}</p>
                        <p>DewPoint: {dayInfo.dewPoint}</p>
                        <p>Intensity: {dayInfo.precipIntensity}</p>
                        <p>Probability: {Math.round(dayInfo.precipProbability * 100)}%</p>
                        <p>Intensity Max: {dayInfo.precipIntensityMax}</p>
                        <p>Intensity Max time: {this.getTime(dayInfo.precipIntensityMaxtime)}</p>
                        <p>Pressure: {dayInfo.pressure}</p>
                    </div>
                </div>

                {data ? <div><Hourly weatherData={this.state.weatherData}/></div> : null}
            </Link>
        );
    }
}
