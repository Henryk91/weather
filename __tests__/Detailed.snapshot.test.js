
import Detailed from '../src/client/views/Components/Detailed/Detailed';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import { weatherData } from "../testData/weatherData.sample"

test('Detailed component renders all details', () => {

    const rendered = renderer.create(
        <Router>
                <Detailed weatherData={weatherData()} />
        </Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
});