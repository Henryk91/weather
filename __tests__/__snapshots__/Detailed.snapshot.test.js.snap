// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detailed component renders all details 1`] = `
<a
  href="/"
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  <h1>
    Fri Mar 08 2019
  </h1>
  <canvas
    height={60}
    width={60}
  />
  <p
    className="yellowP"
  >
    Foggy in the morning.
  </p>
  <p>
    (click anywhere to go back)
  </p>
  <div
    className="bigScreen detailed"
  >
    <div>
      <p
        className="leadP"
      >
        Wind
      </p>
      <p>
        Bearing: 
        76
      </p>
      <p>
        Gust: 
        4.56
      </p>
      <p>
        Gust Time: 
        17:00
      </p>
      <p>
        Speed: 
        1.74
      </p>
    </div>
    <div>
      <p
        className="leadP"
      >
        Sky
      </p>
      <p>
        Sunrise: 
        05:55
         
      </p>
      <p>
        Sunset: 
        18:21
      </p>
      <p>
        UV Index: 
        10
      </p>
      <p>
        UV Index Time: 
        12:00
      </p>
      <p>
        Visibility: 
        10.61
      </p>
      <p>
        Ozone: 
        255.49
      </p>
    </div>
    <div>
      <hr />
      <p
        className="leadP"
      >
        Temp
      </p>
      <p>
        High: 
        30.08
      </p>
      <p>
        High Time: 
        15:00
      </p>
      <p>
        Low: 
        18.04
      </p>
      <p>
        Low Time: 
        05:00
      </p>
      <p>
        Max: 
        30.08
      </p>
      <p>
        Max Time: 
        15:00
      </p>
      <p>
        Min: 
        18.76
      </p>
      <p>
        Min Time: 
        04:00
      </p>
    </div>
    <div>
      <hr />
      <p
        className="leadP"
      >
        Precipitation
      </p>
      <p>
        Cloud Cover: 
        40
      </p>
      <p>
        Humidity: 
        0.67
      </p>
      <p>
        DewPoint: 
        16.67
      </p>
      <p>
        Intensity: 
        0.0635
      </p>
      <p>
        Probability: 
        18
        %
      </p>
      <p>
        Intensity Max: 
        0.5156
      </p>
      <p>
        Intensity Max time: 
        
      </p>
      <p>
        Pressure: 
        1013.39
      </p>
    </div>
  </div>
  <div>
    <div>
      <h1>
        Hourly Forcast
      </h1>
      <div
        className="weatherLine"
      >
        <p>
          19:00
           
          Fri Mar 08 
        </p>
        <p>
          24.73
          °
        </p>
        <p>
          Wind: 
          3.33
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          20:00
           
          Fri Mar 08 
        </p>
        <p>
          23.02
          °
        </p>
        <p>
          Wind: 
          2.97
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          21:00
           
          Fri Mar 08 
        </p>
        <p>
          22.15
          °
        </p>
        <p>
          Wind: 
          2.65
        </p>
        <p>
          Persip: 
          4
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          22:00
           
          Fri Mar 08 
        </p>
        <p>
          21.88
          °
        </p>
        <p>
          Wind: 
          1.82
        </p>
        <p>
          Persip: 
          5
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          23:00
           
          Fri Mar 08 
        </p>
        <p>
          21.61
          °
        </p>
        <p>
          Wind: 
          1.25
        </p>
        <p>
          Persip: 
          8
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          00:00
           
          Sat Mar 09 
        </p>
        <p>
          21.22
          °
        </p>
        <p>
          Wind: 
          1.03
        </p>
        <p>
          Persip: 
          8
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          01:00
           
          Sat Mar 09 
        </p>
        <p>
          20.96
          °
        </p>
        <p>
          Wind: 
          0.77
        </p>
        <p>
          Persip: 
          8
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          02:00
           
          Sat Mar 09 
        </p>
        <p>
          20.58
          °
        </p>
        <p>
          Wind: 
          0.6
        </p>
        <p>
          Persip: 
          8
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          03:00
           
          Sat Mar 09 
        </p>
        <p>
          19.57
          °
        </p>
        <p>
          Wind: 
          0.5
        </p>
        <p>
          Persip: 
          7
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          04:00
           
          Sat Mar 09 
        </p>
        <p>
          18.41
          °
        </p>
        <p>
          Wind: 
          0.53
        </p>
        <p>
          Persip: 
          4
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          05:00
           
          Sat Mar 09 
        </p>
        <p>
          18.04
          °
        </p>
        <p>
          Wind: 
          0.66
        </p>
        <p>
          Persip: 
          4
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          06:00
           
          Sat Mar 09 
        </p>
        <p>
          18.96
          °
        </p>
        <p>
          Wind: 
          0.59
        </p>
        <p>
          Persip: 
          3
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          07:00
           
          Sat Mar 09 
        </p>
        <p>
          20.4
          °
        </p>
        <p>
          Wind: 
          0.65
        </p>
        <p>
          Persip: 
          3
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          08:00
           
          Sat Mar 09 
        </p>
        <p>
          22.16
          °
        </p>
        <p>
          Wind: 
          0.84
        </p>
        <p>
          Persip: 
          3
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          09:00
           
          Sat Mar 09 
        </p>
        <p>
          24.39
          °
        </p>
        <p>
          Wind: 
          1.09
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          10:00
           
          Sat Mar 09 
        </p>
        <p>
          26.61
          °
        </p>
        <p>
          Wind: 
          1.44
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          11:00
           
          Sat Mar 09 
        </p>
        <p>
          28.11
          °
        </p>
        <p>
          Wind: 
          1.78
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          12:00
           
          Sat Mar 09 
        </p>
        <p>
          29.59
          °
        </p>
        <p>
          Wind: 
          2.11
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          13:00
           
          Sat Mar 09 
        </p>
        <p>
          30.61
          °
        </p>
        <p>
          Wind: 
          2.42
        </p>
        <p>
          Persip: 
          1
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          14:00
           
          Sat Mar 09 
        </p>
        <p>
          31.14
          °
        </p>
        <p>
          Wind: 
          2.61
        </p>
        <p>
          Persip: 
          1
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          15:00
           
          Sat Mar 09 
        </p>
        <p>
          31.06
          °
        </p>
        <p>
          Wind: 
          2.57
        </p>
        <p>
          Persip: 
          1
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          16:00
           
          Sat Mar 09 
        </p>
        <p>
          30.58
          °
        </p>
        <p>
          Wind: 
          2.41
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          17:00
           
          Sat Mar 09 
        </p>
        <p>
          29.66
          °
        </p>
        <p>
          Wind: 
          2.17
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          18:00
           
          Sat Mar 09 
        </p>
        <p>
          28.03
          °
        </p>
        <p>
          Wind: 
          1.9
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          19:00
           
          Sat Mar 09 
        </p>
        <p>
          25.96
          °
        </p>
        <p>
          Wind: 
          1.59
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          20:00
           
          Sat Mar 09 
        </p>
        <p>
          24.29
          °
        </p>
        <p>
          Wind: 
          1.35
        </p>
        <p>
          Persip: 
          0
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          21:00
           
          Sat Mar 09 
        </p>
        <p>
          23.34
          °
        </p>
        <p>
          Wind: 
          1.16
        </p>
        <p>
          Persip: 
          2
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          22:00
           
          Sat Mar 09 
        </p>
        <p>
          22.72
          °
        </p>
        <p>
          Wind: 
          1.01
        </p>
        <p>
          Persip: 
          3
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          23:00
           
          Sat Mar 09 
        </p>
        <p>
          22.2
          °
        </p>
        <p>
          Wind: 
          0.9
        </p>
        <p>
          Persip: 
          5
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          00:00
           
          Sun Mar 10 
        </p>
        <p>
          21.61
          °
        </p>
        <p>
          Wind: 
          0.84
        </p>
        <p>
          Persip: 
          6
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          01:00
           
          Sun Mar 10 
        </p>
        <p>
          21.24
          °
        </p>
        <p>
          Wind: 
          0.84
        </p>
        <p>
          Persip: 
          9
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          02:00
           
          Sun Mar 10 
        </p>
        <p>
          21.04
          °
        </p>
        <p>
          Wind: 
          0.85
        </p>
        <p>
          Persip: 
          12
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          03:00
           
          Sun Mar 10 
        </p>
        <p>
          21
          °
        </p>
        <p>
          Wind: 
          0.87
        </p>
        <p>
          Persip: 
          13
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          04:00
           
          Sun Mar 10 
        </p>
        <p>
          21.07
          °
        </p>
        <p>
          Wind: 
          0.9
        </p>
        <p>
          Persip: 
          13
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          05:00
           
          Sun Mar 10 
        </p>
        <p>
          21.27
          °
        </p>
        <p>
          Wind: 
          0.96
        </p>
        <p>
          Persip: 
          15
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          06:00
           
          Sun Mar 10 
        </p>
        <p>
          21.41
          °
        </p>
        <p>
          Wind: 
          1.05
        </p>
        <p>
          Persip: 
          16
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          07:00
           
          Sun Mar 10 
        </p>
        <p>
          21.48
          °
        </p>
        <p>
          Wind: 
          1.17
        </p>
        <p>
          Persip: 
          20
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          08:00
           
          Sun Mar 10 
        </p>
        <p>
          22.33
          °
        </p>
        <p>
          Wind: 
          1.27
        </p>
        <p>
          Persip: 
          19
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          09:00
           
          Sun Mar 10 
        </p>
        <p>
          24.34
          °
        </p>
        <p>
          Wind: 
          1.35
        </p>
        <p>
          Persip: 
          13
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          10:00
           
          Sun Mar 10 
        </p>
        <p>
          26.92
          °
        </p>
        <p>
          Wind: 
          1.43
        </p>
        <p>
          Persip: 
          7
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          11:00
           
          Sun Mar 10 
        </p>
        <p>
          28.35
          °
        </p>
        <p>
          Wind: 
          1.54
        </p>
        <p>
          Persip: 
          7
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          12:00
           
          Sun Mar 10 
        </p>
        <p>
          29.05
          °
        </p>
        <p>
          Wind: 
          1.75
        </p>
        <p>
          Persip: 
          13
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          13:00
           
          Sun Mar 10 
        </p>
        <p>
          28.87
          °
        </p>
        <p>
          Wind: 
          1.98
        </p>
        <p>
          Persip: 
          16
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          14:00
           
          Sun Mar 10 
        </p>
        <p>
          28.1
          °
        </p>
        <p>
          Wind: 
          2.06
        </p>
        <p>
          Persip: 
          22
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          15:00
           
          Sun Mar 10 
        </p>
        <p>
          27.12
          °
        </p>
        <p>
          Wind: 
          1.91
        </p>
        <p>
          Persip: 
          23
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          16:00
           
          Sun Mar 10 
        </p>
        <p>
          26.46
          °
        </p>
        <p>
          Wind: 
          1.63
        </p>
        <p>
          Persip: 
          25
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          17:00
           
          Sun Mar 10 
        </p>
        <p>
          25.63
          °
        </p>
        <p>
          Wind: 
          1.38
        </p>
        <p>
          Persip: 
          27
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          18:00
           
          Sun Mar 10 
        </p>
        <p>
          24.81
          °
        </p>
        <p>
          Wind: 
          1.19
        </p>
        <p>
          Persip: 
          30
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
      <div
        className="weatherLine"
      >
        <p>
          19:00
           
          Sun Mar 10 
        </p>
        <p>
          23.7
          °
        </p>
        <p>
          Wind: 
          1.04
        </p>
        <p>
          Persip: 
          34
          %
        </p>
        <canvas
          height={40}
          width={40}
        />
      </div>
    </div>
  </div>
</a>
`;
